p4_symbolic/symbolic/symbolic.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "p4_symbolic/symbolic/symbolic.h"
+
+#include "p4_symbolic/symbolic/table.h"
+#include "p4_symbolic/symbolic/util.h"
+
+namespace p4_symbolic {
+namespace symbolic {
+
+z3::context *Z3_CONTEXT = nullptr;
```

Comment:
Can we just initialize this variable once, and keep it forever? How the code initializes and frees it right now, is hard to follow for me.